WIP: Upgrade to version 1.18.0

quaresmajose opened the pull request:
This pull request It was also submitted to OE-Core
https://patchwork.openembedded.org/series/26619

I have marked this as WIP because it has not yet been accepted in OE-Core,
It can be useful for someone and can also be tested by a larger group of users

quaresmajose committed:
gstreamer1.0: upgrade to version 1.18.0

    * 0001-gst-gstpluginloader.c-when-env-var-is-set-do-not-fal.patch
      Submited
      https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/669

    * 0002-meson-build-gir-even-when-cross-compiling-if-introsp.patch
      Removed since these are included in 1.18.0

    * 0003-meson-Add-valgrind-feature.patch
      Removed because upstream don't depend anymore on valgrind
      https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/595

    * 0002-Remove-unused-valgrind-detection.patch
      Upstream-Status: Backport [a2cbf75523cdf8a4df1baa7007d86ef455972245]
      https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/a2cbf75523cdf8a4df1baa7007d86ef455972245

    API and plugin documentation are no longer built with gtk_doc.
    The gtk_doc documentation has been removed in favour of a new unified
    documentation module built with hotdoc.
    Distributors should use the documentation release tarball instead of trying
    to package hotdoc and building the documentation from scratch.

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-plugins-base: upgrade to version 1.18.0

    * 0001-meson-build-gir-even-when-cross-compiling-if-introsp.patch
    * 0002-meson-Add-variables-for-gir-files.patch
      Removed since these are included in 1.18.0

License-Update: remove file://common/coverage/coverage-report.pl
Upstream dont have it any more.

    API and plugin documentation are no longer built with gtk_doc.
    The gtk_doc documentation has been removed in favour of a new unified
    documentation module built with hotdoc.
    Distributors should use the documentation release tarball instead of trying
    to package hotdoc and building the documentation from scratch.

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-plugins-good: upgrade to version 1.18.0

    * 0001-qmlgl-ensure-Qt-defines-GLsync-to-fix-compile-on-som.patch
      Removed since these are included in 1.18.0

License-Update: remove file://common/coverage/coverage-report.pl
Upstream dont have it any more.

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-plugins-bad: upgrade to version 1.18.0

    * 0001-meson-build-gir-even-when-cross-compiling-if-introsp.patch
      Removed since these are included in 1.18.0

PACKAGECONFIG:
 - vdpau has been removed upstream
 - vulkan needs now needs shaderc-native to build

EXTRA_OEMESON:
 - yadif has been removed upstream
 - nvdec and nvenc has been replaced with nvcodec

FILES:
 - add new files for transcode plugin

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-plugins-ugly: upgrade to version 1.18.0

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-python: upgrade to version 1.18.0

    * 0001-meson.build-fix-builds-with-python-3.8.patch
      Removed since these are included in 1.18.0

DEPENDS:
RDEPENDS:
 - Upstream now depends gstreamer1.0-plugins-base

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-rtsp-server: upgrade to version 1.18.0

    * 0001-meson-build-gir-even-when-cross-compiling-if-introsp.patch
      Removed since these are included in 1.18.0

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-vaapi: upgrade to version 1.18.0

    * 0001-vaapsink-downgrade-to-marginal.patch
      Removed since these are included in 1.18.0
      https://gitlab.freedesktop.org/gstreamer/gstreamer-vaapi/-/commit/13d090b5660a2e2e2c997db44ba9a391101e8bd6

    API and plugin documentation are no longer built with gtk_doc.
    The gtk_doc documentation has been removed in favour of a new unified
    documentation module built with hotdoc.
    Distributors should use the documentation release tarball instead of trying
    to package hotdoc and building the documentation from scratch.

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gst-examples: upgrade to version 1.18.0

DEPENDS:
 - Upstream now depends on libsoup-2.4 and json-glib

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gst-devtools: add version 1.18.0 (gst-validate -> gst-devtools)

The gst-validate tarball has been superseded by the gst-devtools tarball for consistency with the git module name.

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-omx: upgrade to version 1.18.0

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-libav: upgrade to version 1.18.0

License-Update: All code is now LGPL-2.1+

Starting with 1.17, we will not ship a copy of FFmpeg in our release
tarballs, and all the remaining code is LGPL2.
https://gitlab.freedesktop.org/gstreamer/gst-libav/-/commit/c437e175ad9c82732ac46e36ccc398ad041bf13b

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
spirv-tools: enable native build and install more header files

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
glslang: add receipe

Glslang is the official reference compiler front end for the
OpenGL ES and OpenGL shading languages. It implements a strict interpretation
of the specifications for these languages. It is open and free for anyone to use,
either from a command line or programmatically.

This receipe is needed to build the new shaderc

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
shaderc: add receipe

A collection of tools, libraries and tests for shader compilation

This receipe is needed to build the gstreamer vulkan plugin
as it provides the binary glslc.

It is based on arch linux shaderc packge including the patches
https://github.com/archlinux/svntogit-packages/blob/91f0fa6ee3a220264a448527c1a8cf037caaad8e/trunk/PKGBUILD

    * 0001-fix-glslang-link-order.patch
      Upstream-Status: Backport [21c8be385b3fab5edcb934a6d99f69fd389c4e67]

    * 0002-shaderc-2019.0-fix-build-against-new-glslang.patch
      Upstream-Status: Pending

    * 0003-cmake-de-vendor-libs-and-disable-git-versioning.patch
      Upstream-Status: Inappropriate [configuration]

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
orc: Upgrade 0.4.29 -> 0.4.32

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

otavio commented:
Awesome @quaresmajose; I marked @dv1 and @fbertux to review it as well.

otavio commented on recipes-multimedia/gstreamer/gstreamer1.0_1.18.0.bb:
```diff
@@ -6,41 +6,38 @@ BUGTRACKER = "https://bugzilla.gnome.org/enter_bug.cgi?product=Gstreamer"
                    debug \
-                   tools"
+                   tools \
+                   capabilities"
```
The capabilities addition should be added as another patch as it is changing the current build settings.

quaresmajose replied:
I have mention this change on the commit message
```
    * capfix.patch
      Removed as the new PACKAGECONFIG capabilities can be used instead to achieve the same goals.
      Fix reproducibility issue around libcap.
```
But after review this again it has a different behavior in this patch set

On the 1.16 it uses 'setuid-root'` if the 'setcap' is not provided in PACKAGECONFIG
I will need to do the same behavior and remove the new 'capabilities' from default PACKAGECONFIG

quaresmajose replied:
I can remove the PACKAGECONFIG `capabilities` and use the  `setcap`  to do the same.
With this changes I only drop the patch `capfix.patch` and adjusts the `setcap`.

```patch
-PACKAGECONFIG[setcap] = "-Dsetcap=enabled,-Dsetcap=disabled,libcap libcap-native"
+PACKAGECONFIG[setcap] = ",,libcap libcap-native"
```

quaresmajose replied:
I also remove `libcap` from `DEPENDS` because it is only needed to set the capabilities.
Now it as the same behavior of the version 1.16.

otavio commented on recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb:
```diff
@@ -1,19 +1,16 @@
 
 LICENSE = "GPLv2+ & LGPLv2.1+"
 LIC_FILES_CHKSUM = "file://COPYING;md5=a6f89e2100d9b6cdffcea4f398e37343 \
-                    file://common/coverage/coverage-report.pl;beginline=2;endline=17;md5=a4e1830fce078028c8f0974161272607 \
```
This removal ought to be noted on the commit log.

quaresmajose replied:
I will fix this

otavio commented on recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb:
```diff
@@ -23,13 +20,15 @@ RPROVIDES_${PN}-soup += "${PN}-souphttpsrc"
 PACKAGECONFIG ??= " \
     ${GSTREAMER_ORC} \
     ${@bb.utils.filter('DISTRO_FEATURES', 'pulseaudio x11', d)} \
+    ${@bb.utils.contains('TUNE_FEATURES', 'm64', 'asm', '', d)} \
```
Please, if this is required for the new version, please add to the commit. Otherwise, if it is an improvement, it ought to be done as another commit.

quaresmajose replied:
It is a new meson option on this version.
I will mention this on the commit message

otavio commented on recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb:
```diff
@@ -62,6 +62,7 @@ EXTRA_OEMESON += " \
     -Doss4=disabled \
     -Dosxaudio=disabled \
     -Dosxvideo=disabled \
+    -Drpicamsrc=disabled \
```
New commit, please.

quaresmajose replied:
I will do it

otavio commented on recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_1.18.0.bb:
```diff
@@ -93,18 +91,22 @@ PACKAGECONFIG[zbar]            = "-Dzbar=enabled,-Dzbar=disabled,zbar"
     -Dipcpipeline=enabled \
     -Dnetsim=enabled \
     -Dshm=enabled \
+    -Dv4l2codecs=enabled \
```
Those ought to be new PACKAGECONFIG options, no? also, in a new commit.

quaresmajose replied:
<del>This don't have any dependencies on external packages.

I will add a `PACKAGECONFIG` for this because it depends on `libgudev`

otavio commented on recipes-multimedia/gstreamer/gstreamer1.0-vaapi_1.18.0.bb:
```diff
@@ -8,28 +8,24 @@ REALPN = "gstreamer-vaapi"
 EXTRA_OEMESON += " \
+    -Ddoc=disabled \
     -Dexamples=disabled \
+    -Dtests=disabled \
```
This seems to break the tests package, below.

quaresmajose replied:
probably yes, I will enable it

otavio commented on recipes-multimedia/gstreamer/gst-examples_1.18.0.bb:
```diff
@@ -2,14 +2,14 @@ SUMMARY = "GStreamer examples (including gtk-play, gst-play)"
 LIC_FILES_CHKSUM = "file://playback/player/gtk/gtk-play.c;beginline=1;endline=20;md5=f8c72dae3d36823ec716a9ebcae593b9"
 
-DEPENDS = "glib-2.0 gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-bad gtk+3 glib-2.0-native"
+DEPENDS = "glib-2.0 gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-bad gtk+3 libsoup-2.4 json-glib glib-2.0-native"
```
Please mention new dependencies on the commit log.

quaresmajose replied:
I will do it

quaresmajose committed:
gstreamer1.0-plugins-good: disable new meson options

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0: Fix reproducibility issue around libcap

Currently gstreamer configuration depends libcap and on whether
setcap is found on the host system.

Removing libcap from DEPENDS and only use it when the 'setcap' is enabled.

    * 0004-capfix.patch
      Removed as the same goals can be achieved only with the PACKAGECONFIG 'setcap'

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-plugins-base: add new meson option as PACKAGECONFIG

- gl_winsys have a new meson option to set egl

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-plugins-good: add new meson option as PACKAGECONFIG

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-plugins-bad: disable new meson options

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-plugins-bad: add new meson options as PACKAGECONFIG

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-python: install append is not need any more

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

quaresmajose committed:
gstreamer1.0-plugins-good: on wayland qt5 needs qtwayland

The code in ext/qt/meson.build looks for the Qt5WaylandClient pkg-config file.
Without this module, qmlglsink will not work with Wayland.
This can be tested by checking whether or not HAVE_QT_WAYLAND has
been added to the cflags in the autogenerated ninja file.

Writen by:
Carlos Rafael Giani
https://lists.openembedded.org/g/openembedded-core/message/143574

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>
